src/Package.php

```diff
@@ -41,76 +41,74 @@ class Package
      * $package = Package::new();
      *
      * add_action(
      *      $package->hookName(Package::ACTION_INIT),
-     *      $callback
+     *      fn (Package $package) => // do something,
      * );
      * </code>
      */
     public const ACTION_INIT = 'init';
 
     /**
-     * Custom action which is triggered after the application
-     * is booted to access container and properties.
+     * Very similar to `ACTION_INIT`, but it is static, so not dependent on package name.
+     * It passes package name as first argument.
      *
      * @example
-     * <code>
-     * $package = Package::new();
-     *
-     * add_action(
-     *      $package->hookName(Package::ACTION_READY),
-     *      $callback
-     * );
-     * </code>
+     *  <code>
+     *  add_action(
+     *       Package::ACTION_MODULARITY_INIT,
+     *       fn (string $packageName, Package $package) => // do something,
+     *       10,
+     *       2
+     *  );
+     *  </code>
      */
-    public const ACTION_READY = 'ready';
+    public const ACTION_MODULARITY_INIT = self::HOOK_PREFIX . self::ACTION_INIT;
 
     /**
-     * Custom action which is triggered when a failure happens during the building stage.
-     *
-     * @example
-     * <code>
-     * $package = Package::new();
-     *
-     * add_action(
-     *      $package->hookName(Package::ACTION_FAILED_BUILD),
-     *      $callback
-     * );
-     * </code>
+     * Action fired when it is safe to access container.
+     * Add more modules is not anymore possible at this stage.
+     */
+    public const ACTION_INITIALIZED = 'initialized';
+
+    /**
+     * Action fired when plugin finished its bootstrapping process, all its hooks are added.
+     * Add more modules is not anymore possible at this stage.
+     */
+    public const ACTION_BOOTED = 'ready';
+
+    /**
+     * Action fired when anything went wrong during the "build" procedure.
      */
     public const ACTION_FAILED_BUILD = 'failed-build';
 
     /**
-     * Custom action which is triggered when a failure happens during the booting stage.
-     *
-     * @example
-     * <code>
-     * $package = Package::new();
-     *
-     * add_action(
-     *      $package->hookName(Package::ACTION_FAILED_BOOT),
-     *      $callback
-     * );
-     * </code>
+     * Action fired when anything went wrong during the "boot" procedure.
      */
     public const ACTION_FAILED_BOOT = 'failed-boot';
 
     /**
-     * Custom action which is triggered when a package is connected.
+     * Action fired when adding a module failed.
      */
-    public const ACTION_PACKAGE_CONNECTED = 'package-connected';
+    public const ACTION_FAILED_ADD_MODULE = 'failed-add-module';
 
     /**
-     * Custom action which is triggered when a package cannot be connected.
+     * Action fired when a package connection failed.
      */
-    public const ACTION_FAILED_CONNECTION = 'failed-connection';
+    public const ACTION_FAILED_CONNECT = 'failed-connect';
```

Comment:
I don't see a back-compat alias for this **public** class constant, and the value has also changed.

Reply:
My mistake. I thought for some reason the constant was introduced in the PR, but it was already there. So yes, we need a back-compat alias. Thanks.